### PR TITLE
SIMPLY-2502 DPLA Audiobook DRM check

### PR DESCRIPTION
**Pull request by ErnestFan**
**What's this do?**
Retrieve Feedbook certificate from Keychain
Generate signature and compare it the signature value within the manifest
Minimum iOS version is bumped to 10.0 

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SIMPLY-2502

**How should this be tested? / Do these changes have associated tests?**
Testings will be on SimplyE

**Dependencies for merging? Releasing to production?**
No

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@ErnestFan

### CODE REVIEW

**Comment by ettore** [comment]
@ErnestFan can you make a note in the ticket and tag Risa that this change requires us to drop iOS 9 support?

**Comment by ettore** [comment]
one more thing: you should probably merge this to a feature branch? Either that or add a version tag and bind SimplyE to that tag. I think currently it just refers to master.

File: NYPLAudiobookToolkit/NYPLAudiobookToolkit.h [hunk 1]
```diff
@@ -14,4 +14,7 @@ FOUNDATION_EXPORT double NYPLAudiobookToolkitVersionNumber;
 //! Project version string for NYPLAudiobookToolkit.
 FOUNDATION_EXPORT const unsigned char NYPLAudiobookToolkitVersionString[];
 
+//! For storing and accessing certificate from Keychain
+FOUNDATION_EXPORT NSString *const FeedbookDRMCertificateTag;
```
[the remark this answers is not in the document]
Comment (ErnestFan) [reply NYPLAudiobookToolkit/NYPLAudiobookToolkit.h:18]: I am storing the cert on the keychain within `NYPLKeychainManager`, similar to `manageFeedbooksData()`.
I can move that into the NYPLAudiobook tho

File: NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift [hunk 2]
```diff
@@ -54,7 +54,89 @@ class FeedbookDRMProcessor {
...
+            let blockSize = SecKeyGetBlockSize(privateSecKey)
+            
+            guard Int(CC_SHA256_DIGEST_LENGTH) <= blockSize - 11 else {
+                ATLog(.error, "Invalid data size, data size cannot be larger or equal to key size - 11 bytes")
+                // ref: https://developer.apple.com/documentation/security/1618025-seckeyrawsign
+                return true
+            }
+            
+            var digestBytes = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))
+            RSAUtils.SHA256HashedData(from: (licenseData as NSData)).getBytes(&digestBytes, length: Int(CC_SHA256_DIGEST_LENGTH))
```
[the remark this answers is not in the document]
Comment (ErnestFan) [reply NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift:112]: `getBytes` is called on the return value of `SHA256HashedData`, and the bytes are being stored in `digestBytes`

File: NYPLAudiobookToolkit/NYPLAudiobookToolkit.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ettore) [reply NYPLAudiobookToolkit/NYPLAudiobookToolkit.h:18]: oh I see, you're doing it on the SimplyE / app side. That's fine, actually probably better. I wasn't understanding.

File: NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (ettore) [reply NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift:112]: ohhh I missed the chaining. sorry. I read it as the `getBytes` result was a param to `SHA256HashedData`

**Comment by ErnestFan** [comment]
> one more thing: you should probably merge this to a feature branch? Either that or add a version tag and bind SimplyE to that tag. I think currently it just refers to master.

Created a `dpla/develop` branch for this, I will do the same for the SimplyE PR later on

File: NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift [hunk 3]
```diff
@@ -53,8 +53,92 @@ class FeedbookDRMProcessor {
...
+        }
+        
+        var licenseDocument = manifest
+        licenseDocument["metadata"] = metadata
+
+        return verifySignature(forLicenseDoc: licenseDocument, signatureValue: signatureValue)
+    }
+    
+    // Verify the signature within the manifest using the private key from Keychain
+    // @param the manifest(license document) without the signature
```
Comment (ettore) [inline NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift:70]: docs in swift use markdown, not the old objc `@param` syntax

File: NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift [hunk 4]
```diff
@@ -53,8 +53,92 @@ class FeedbookDRMProcessor {
...
+            let privateSecKeyProperties = [
+                kSecAttrKeyType: kSecAttrKeyTypeRSA,
+                kSecAttrKeyClass: kSecAttrKeyClassPrivate
+            ]
+
+            guard let privateSecKey = SecKeyCreateWithData(privateKeyData as NSData,
+                                                           privateSecKeyProperties as NSDictionary,
+                                                           &error) else {
+                ATLog(.error, "Failed to create SecKey from private key - \(error)")
+                return true
```
Comment (ettore) [inline NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift:97]: shouldn't we return `false` here?

File: NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift [hunk 5]
```diff
@@ -53,8 +53,92 @@ class FeedbookDRMProcessor {
...
+            let status = SecKeyRawSign(privateSecKey,
+                                       .PKCS1SHA256,
+                                       digestBytes,
+                                       digestBytes.count,
+                                       &signatureBytes,
+                                       &signatureDataLength)
+            
+            guard status == noErr else {
+                ATLog(.error, "Failed to sign data - \(status.description)")
+                return true
```
Comment (ettore) [inline NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift:128]: this should also be `false`, right?

**Pull request merged**